addon/v1alpha1/types_managedclusteraddon.go

```diff
@@ -280,7 +268,134 @@ const (
 	// HostedManifestLocationHostingValue is a value of the annotation HostedManifestLocationAnnotationKey,
 	// indicates the manifest will be deployed on the hosting cluster in Hosted mode.
 	HostedManifestLocationHostingValue = "hosting"
-	// HostedManifestLocationNoneValue is a value of the annotation HostedManifestLocationAnnotationKey,,
+	// HostedManifestLocationNoneValue is a value of the annotation HostedManifestLocationAnnotationKey,
 	// indicates the manifest will not be deployed in Hosted mode.
 	HostedManifestLocationNoneValue = "none"
+
+	// Finalizers on the managedClusterAddon.
+
+	// AddonDeprecatedPreDeleteHookFinalizer is the finalizer for an addon which has deployed hook objects.
+	// Deprecated: please use the AddonPreDeleteHookFinalizer instead.
+	AddonDeprecatedPreDeleteHookFinalizer = "cluster.open-cluster-management.io/addon-pre-delete"
+	// AddonDeprecatedHostingPreDeleteHookFinalizer is the finalizer for an addon which has deployed hook objects
+	// on hosting cluster.
+	// Deprecated: please use the AddonHostingPreDeleteHookFinalizer instead.
+	AddonDeprecatedHostingPreDeleteHookFinalizer = "cluster.open-cluster-management.io/hosting-addon-pre-delete"
+	// AddonDeprecatedHostingManifestFinalizer is the finalizer for an addon which has deployed manifests on the external
+	// hosting cluster in Hosted mode.
+	// Deprecated: please use the AddonHostingPreDeleteHookFinalizer instead.
+	AddonDeprecatedHostingManifestFinalizer = "cluster.open-cluster-management.io/hosting-manifests-cleanup"
+
+	// AddonPreDeleteHookFinalizer is the finalizer for an addon which has deployed hook objects.
+	AddonPreDeleteHookFinalizer = "addon.open-cluster-management.io/addon-pre-delete"
+	// AddonHostingPreDeleteHookFinalizer is the finalizer for an addon which has deployed hook objects
+	// on hosting cluster.
+	AddonHostingPreDeleteHookFinalizer = "addon.open-cluster-management.io/hosting-addon-pre-delete"
+	// AddonHostingManifestFinalizer is the finalizer for an addon which has deployed manifests on the external
+	// hosting cluster in Hosted mode.
+	AddonHostingManifestFinalizer = "addon.open-cluster-management.io/hosting-manifests-cleanup"
+)
+
+const (
+	// AddonInstallModeBuiltinValueKey is the key of the build in value to represent the addon install mode, addon developers
+	// can use this built in value in manifests.
```

Comment:
Is the value they put in helm chart? I do not think we should put this in api repo if it is not set in any resources.

Reply:
yes, removed.